src/sentry/static/sentry/app/views/substances/substances.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import {substancesGet} from 'app/redux/actions/substance';
+import ReactTable from 'react-table';
+import 'react-table/react-table.css';
+import LoadingError from 'app/components/loadingError';
+import LoadingIndicator from 'app/components/loadingIndicator';
+
+class Substances extends React.Component {
+  componentWillMount() {
+    this.props.getSubstances();
+  }
+
+  render() {
+    if (this.props.loading) {
+      return <LoadingIndicator />;
+    } else if (this.props.errorMessage) {
+      return <LoadingError />;
+    }
+
+    return (
+      <div>
+        <ReactTable
+          data={this.props.substances}
+          columns={[
+            {
+              Header: 'Sample name',
+              accessor: 'name',
+            },
+            {
+              Header: 'Position',
+              id: 'position',
+              accessor: d => d.position.index,
+            },
+            {
+              Header: 'Volume',
+              id: 'volume',
+              accessor: d => d.properties.volume,
+            },
+            {
+              Header: 'Sample Type',
+              id: 'sample_type',
+              accessor: d => d.properties.sample_type,
+            },
+            {
+              Header: 'Priority',
+              id: 'priority',
+              accessor: d => d.priority,
+            },
+            {
+              Header: 'Waiting',
+              id: 'days_waiting',
+              accessor: d => d.days_waiting,
+              Cell: row => (
+                <span>
```

Comment:
This should be a react component if we decide to use it, making the code more readable - again, just getting the preview out directly.

Reply:
:+1: Pephaps add a TODO in the code to indicate this_

Reply:
Will add that!